gobuffalo/azuredeploy.json

```diff
@@ -0,0 +1,196 @@
+{
+    "$schema": "https://schema.management.azure.com/schemas/2014-04-01-preview/deploymentTemplate.json#",
+    "contentVersion": "1.0.0.0",
+    "parameters": {
+      "imageName" : {
+        "type": "String",
+        "defaultValue": "appsvc/sample-hello-world:latest",
+        "metadata": {
+          "description": "The image which defines the application to be hosted in Azure."
+        }
+      },
+      "name": {
+        "type": "String",
+        "defaultValue": "[concat('site', uniqueString(resourceGroup().id, deployment().name))]",
+        "metadata": {
+          "description": "A unique name to identify the site, and its relate assets once it is created."
+        }
+      },
+      "database": {
+        "type": "String",
+        "defaultValue": "none",
+        "allowedValues": [
+            "none",
+            "postgresql",
+            "mysql"
+        ],
+        "metadata": {
+          "description": "The flavor of database that should be deployed alongside this application."
+        }
+      },
+      "databaseName": {
+        "type": "String",
+        "defaultValue": "buffalo_development",
+        "metadata": {
+          "description": "The name that will identify the database which is created, should one be created."
+        }
+      },
+      "databaseAdministratorLogin": {
+          "type": "String",
+          "defaultValue": "[concat('admin', parameters('name'))]",

```

Comment:
remove defaultValue, increases surface area

Reply:
Would that force people to provide a value for the databaseAdministratorLogin, even when they were using "none" for their database option?

Reply:
you can use an empty string for that scenario